src/main/java/com/rcv/Logger.java

```diff
@@ -35,72 +35,117 @@
-  private static final String DEFAULT_FILE_NAME = "rcv.log";
+  private static final String EXECUTION_LOG_FILE_NAME = "rcv.log";
+  // first value here is bytes per MB and the second is max MB for the log file
+  private static final Integer EXECUTION_LOG_FILE_MAX_SIZE_BYTES = 1000000 * 100;
+  // how many execution files to keep
+  private static final Integer EXECUTION_LOG_FILE_COUNT = 2;
+
 
   // function: setup
   // purpose: initialize logging module
   // throws: IOException if unable to open output log file
   static void setup() throws IOException {
     // create and cache default logger
-    defaultLogger = java.util.logging.Logger.getLogger("");
+    executionLogger = java.util.logging.Logger.getLogger(EXECUTION_LOGGER_NAME);
     // remove any loggers the system may have installed
-    for (Handler handler : defaultLogger.getHandlers()) {
-      defaultLogger.removeHandler(handler);
+    for (Handler handler : executionLogger.getHandlers()) {
+      executionLogger.removeHandler(handler);
     }
     // logPath is where default file logging is written
     // "user.dir" property is the current working directory, i.e. folder from whence the rcv jar
     // was launched
-    String logPath = Paths.get(System.getProperty("user.dir"), DEFAULT_FILE_NAME).toString();
+    String logPath = Paths.get(System.getProperty("user.dir"), EXECUTION_LOG_FILE_NAME).toString();
     // formatter specifies how logging output lines should appear
     LogFormatter formatter = new LogFormatter();
     // fileHandler writes formatted strings to file
-    FileHandler fileHandler = new FileHandler(logPath, true);
+    FileHandler fileHandler = new FileHandler(logPath,
+        EXECUTION_LOG_FILE_MAX_SIZE_BYTES,
+        EXECUTION_LOG_FILE_COUNT,
+        true);
     fileHandler.setFormatter(formatter);
     // create a consoleHandler to writes formatted strings to console for debugging
     ConsoleHandler consoleHandler = new ConsoleHandler();
     consoleHandler.setFormatter(formatter);
     // add the  handlers
-    defaultLogger.addHandler(consoleHandler);
-    defaultLogger.addHandler(fileHandler);
+    executionLogger.addHandler(consoleHandler);
+    executionLogger.addHandler(fileHandler);
 
     // create and cache the tabulation logger object
-    // whenever a tabulation happens we will add tabulation-specific file handlers here
+    // whenever a tabulation happens we will add tabulation-specific file handlers to it
     tabulationLogger = java.util.logging.Logger.getLogger(TABULATION_LOGGER_NAME);
   }
 
-  // function: info
-  // purpose: log formatted output to console and audit file at INFO level
-  // format: format string into which object params will be formatted
-  // param: obj object to be parsed into format string
-  static void info(String format, Object... obj) {
-    tabulationLogger.info(String.format(format, obj));
+  // logs text to all output loggers
+  static void allLogs(Level level, String format, Object... obj) {
```

Comment:
My intuition is telling me that there's a more elegant way to implement the loggers so we can just make calls like `executionLogger.warn("Blah!")` instead of `Logger.executionLog(Level.WARN, "Blah!")`. In other words, maybe instead of having these four separate functions for logging in various permutations, it makes more sense to maintain four separate loggers with different permutations of handlers?

Reply:
No problem, I'll think about it more and reassess once we get this merged in.

Reply:
I'm gonna keep thinking about it too.  I think at least "auditLog" can simply be a log call to tabulationLogger with level = verbose, and we use level filter to keep that out of the GUI.  This stuff isn't critical, but it's nice to keep it tight :)